### PR TITLE
[JENKINS-35417] Add new selector: Last build with artifacts

### PR DESCRIPTION
**Pull request by crgilman**
Add new selector: Last completed build with artifacts

The new selector will select the last completed build that is known to have artifacts.

### CODE REVIEW

File: src/main/java/hudson/plugins/copyartifact/LastBuildWithArtifactSelector.java [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+
+public class LastBuildWithArtifactSelector extends BuildSelector {
+
+    @DataBoundConstructor
+    public LastBuildWithArtifactSelector() {
+    }
+
+    @Override
+    public boolean isSelectable(Run<?, ?> run, EnvVars env) {
+        return run.getArtifacts().isEmpty() ? false : true;
```
Comment (ikedam) [inline src/main/java/hudson/plugins/copyartifact/LastBuildWithArtifactSelector.java]: `getHasArtifacts` looks better.
https://javadoc.jenkins-ci.org/hudson/model/Run.html#getHasArtifacts--

File: src/main/java/hudson/plugins/copyartifact/LastBuildWithArtifactSelector.java [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+    }
+
+    /**
+     * @deprecated
+     *      here for backward compatibility. Get it from {@link Jenkins#getDescriptor(Class)}
+     */
+    public static /*almost final*/ Descriptor<BuildSelector> DESCRIPTOR;
```
Comment (ikedam) [inline src/main/java/hudson/plugins/copyartifact/LastBuildWithArtifactSelector.java]: As annotated as `deprecated`, this is an old-style implementaion.
You’d better refer the implementation of newer selectors, like [PermalinkBuildSelector](https://github.com/jenkinsci/copyartifact-plugin/blob/master/src/main/java/hudson/plugins/copyartifact/PermalinkBuildSelector.java).
The implementation would get more simple.

File: src/main/resources/hudson/plugins/copyartifact/Messages.properties [hunk 3]
```diff
@@ -14,6 +14,7 @@ CopyArtifact.ParameterizedName=Value references a build parameter, so it cannot
 CopyArtifact.InvalidVariableName=Contains letters not applicable for variable names.
 PermalinkBuildSelector.DisplayName=Specified by permalink
 LastCompletedBuildSelector.DisplayName=Last completed build (ignoring build status)
+LastBuildWithArtifactSelector.DisplayName=Last completed build with artifacts
```
Comment (ikedam) [inline src/main/resources/hudson/plugins/copyartifact/Messages.properties]: No need for “completed”.

File: src/main/java/hudson/plugins/copyartifact/LastBuildWithArtifactSelector.java [hunk 4]
```diff
@@ -0,0 +1,58 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2019
```
Comment (ikedam) [inline src/main/java/hudson/plugins/copyartifact/LastBuildWithArtifactSelector.java]: Add your name, please.

**Comment by crgilman** [comment]
Thanks for the feedback and taking the time to review my pull request.

**Comment by crgilman** [comment]
I noticed I left my temp name for the test in there...going to update it.

**Comment by ikedam** [comment]
I don’t have time to make a new release this month, and please let me hold this request till March.
Sorry.

**Comment by ikedam** [comment]
Link to Jira: 
https://issues.jenkins-ci.org/browse/JENKINS-35417

**Comment by ikedam** [comment]
I’ll release this feature as copyartifact-1.42 in this weekend.

**Comment by crgilman** [comment]
Sounds good. Thank you.

**Comment by ikedam** [comment]
The release of copyartifact-1.42 is held for the request from @crgilman .

**Pull request merged**